services/thermal_mgr/thermal_mgr.c

```diff
@@ -44,17 +44,57 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
 
+  if (thermalMgrQueueHandle == NULL){
+    return ERR_CODE_INVALID_STATE;
+  }
+  else if(event == NULL){
+    return ERR_CODE_INVALID_ARG;
+  }
+  else if (xQueueSend(thermalMgrQueueHandle, (void *) event, (TickType_t) 0) == errQUEUE_FULL){
+    return ERR_CODE_QUEUE_FULL;
+  }
+
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t interrupt;
+  interrupt.type = THERMAL_MGR_EVENT_INTERRUPT;
+  thermalMgrSendEvent(&interrupt);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  float *temp;
+  thermal_mgr_event_t item;
+  error_code_t read_code;
+  
   while (1) {
-    
+
+    if(xQueueReceive(thermalMgrQueueHandle, &item, (TickType_t) 10) == pdTRUE){ 
```

Comment:
this timeout can be made longer since ur expecting the longest period of time gone without receiving a queue message before you know an error happened to be 5 seconds